extension/js/common/composer.ts

```diff
@@ -1383,21 +1390,26 @@ export class Composer {
   }
 
   private getAndDecryptPreviousMessage = async (): Promise<MessageToReplyOrForward | undefined> => {
-    const determined = await this.app.emailProviderDetermineReplyMsgHeaderVariables();
+    this.S.cached('loader').css('display', 'block');
+    Xss.sanitizeAppend(this.S.cached('loader'), `<span>${Ui.spinner('green')}</span>`);
+    const determined = await this.app.emailProviderDetermineReplyMsgHeaderVariables((progress: number) => this.renderGettingPreviousMessageProgress(progress * 0.3));
     if (determined && determined.lastMsgId) {
       try {
-        const { raw } = await Google.gmail.msgGet(this.urlParams.acctEmail, determined.lastMsgId, 'raw');
+        const { raw } = await Google.gmail.msgGet(this.urlParams.acctEmail, determined.lastMsgId, 'raw',
+          (progress: number) => this.renderGettingPreviousMessageProgress(30 + progress * 0.3));
         const message = await Mime.process(Buf.fromBase64UrlStr(raw!));
         const readableBlocks = message.blocks
           .filter(b => b.type === 'encryptedMsg' || b.type === 'plainText' || b.type === 'plainHtml');
+        const encryptedCount = readableBlocks.filter(b => b.type === 'encryptedMsg').length;
         const decryptedAndFormatedContent: string[] = [];
-        for (const block of readableBlocks) {
+        for (const [index, block] of readableBlocks.entries()) {
```

Comment:
Interesting, I haven't seen that before. That may come handy. I usually did a traditional for loop with i++, but this may be better if I really need both.

Reply:
Honestly, I'm using this first time :) 
I don't like traditional `for` statement because it's hard to read.

I'm 100% sure that traditional `for` would be better for productivity but the difference won't be very big (less the 1ms I think).

Reply:
I'm pretty sure the performance difference is meaningless